### PR TITLE
Merge with upstream v2.1 branch

### PR DESCRIPTION
**Pull request by vittorioromeo**
Needed this for my [Open Hexagon](https://github.com/SuperV1234/SSVOpenHexagon/) project, figured you might be interested.

### CODE REVIEW

File: src/lj_err.c [hunk 1]
```diff
@@ -281,20 +465,137 @@ LJ_FUNCA int lj_err_unwind_dwarf(int version, int actions,
     ** it on non-x64 because the interpreter restores all callee-saved regs.
     */
     lj_err_throw(L, errcode);
+#if LJ_TARGET_X64
+#error "Broken build system -- only use the provided Makefiles!"
+#endif
```
Comment (Wohlstand) [inline src/lj_err.c:470]: Ha-ha, a good kick to the ass to fix the CMake build where are new internal flags required to be defined now.

**Comment by Wohlstand** [comment]
Okay, I had to fix the small mistake in the CMake build that was caused by the new addition into the lj_err.c.

My reply to the person who introduced that line: Unfortunately, Makefile isn't portable and very hard to integrate into the thing, also requires a lot of tools to get it being usable. Especially the pain for the cross-compilation. I do integrate the sources package of the library into my project and I keep the option to get it buildable. It's the biggest pain to set up the environment for Visual Studio on Windows when building the project through MSVC, but glad it's much easier when using MinGW builds. I also do like to use the Ninja builder which goes much better than GNU Make, and I do use it if it's available for a certain platform. In some cases, the GNU Make won't be available, and the Ninja just. The even bigger pain appears with Android builds where CMake build is much easier than hacking with the GNU Make builder workarounds.

**Comment by vittorioromeo** [comment]
@Wohlstand: it wasn't me :) 

The original maintainer of LuaJIT seems, unfortunately, pretty close-minded: https://github.com/LuaJIT/LuaJIT/issues/476#issuecomment-562999473

Quite sad.

**Comment by Wohlstand** [comment]
Yeah, they just didn't like an idea to maintain more build systems that they already support. Also, Mike's personal disliking of the CMake building system. Anyway, at my work, we had another build system - the set of python scripts that do generate the MakeFile for the whole project. Yes, they "just work", but they are HUGE PAIN because they aren't flexible and the Makefile builder is often limited and hard to maintain. Most of the pain macOS developers had with that mess, and they all blamed it as the biggest cludge in the universe. However, after months, the leaders of the development department desired to finally move from the hand-made python-based Makefile generator into the CMake. Since the month of the work (the project itself is a huge and complex surveillance software thing), the whole build got moved into the new builder. As a result, it builds almost two times faster than the previous builder (with the same number of parallel processes use) since we generated the Ninja build instead of the Makefile. At my other projects, I had to use QMake, but once upon I found it's limited for me and requires me to produce an extra set of workarounds. Once I replaced it with the CMake, I got much more convenient tools and a simpler building process of my projects.

**Pull request merged**